src/runner.ts

```diff
@@ -4,17 +4,36 @@ import { DocumentInventory } from './documentInventory';
 import { lintNode } from './linter';
 import { getDiagnosticSeverity } from './rulesets/severity';
 import { FunctionCollection, IGivenNode, IRule, IRuleResult, IRunRule, RunRuleCollection } from './types';
+import { RulesetExceptionCollection } from './types/ruleset';
 import { hasIntersectingElement } from './utils/';
+import { IExceptionLocation, pivotExceptions } from './utils/pivotExceptions';
 
 export const isRuleEnabled = (rule: IRule) => rule.severity !== void 0 && getDiagnosticSeverity(rule.severity) !== -1;
 
 export const runRules = (
   documentInventory: DocumentInventory,
   rules: RunRuleCollection,
   functions: FunctionCollection,
+  exceptions: RulesetExceptionCollection,
 ): IRuleResult[] => {
   const results: IRuleResult[] = [];
 
+  // TODO:
+  // Ruleset
+  // - To be investigated: How to cope with STDIN as a source and exception relative locations?
+  // LoadRuleSet
+  // - TODO: Find out how paths that escape out of the root behave
+  // Spectral
+  // - idea: report exceptions hit through a new 'Silenced' Severy level (-2) -> This opens up a wide range of possible user friendly reporting
```

Comment:
This wouldn't play well with our ecosystem :/
Could use `hint` or `debug` perhaps?

Reply:
@P0lip Ok. Another proposal: extend the `IRuleResult` interface to make it expose a `silenced` boolean property. Thoughts?

Reply:
> @P0lip Ok. Another proposal: extend the IRuleResult interface to make it expose a silenced boolean property. Thoughts?

I'm not sure. Perhaps we simply don't log such errors at all?
We could expose a flag such as `--ignore-skipped-diagnostics` or something similar that would be true by default, similarly to what we do with unknown formats.
If one wants to have ignored diagnostics, that flag could be passed and set to false.

What do you think?
cc @philsturgeon - what's your take on that?

Reply:
We are using many different words here all of which have distinct meaning. 

- Diagnostics - this is a new term, lets avoid introducing it.
- Violations - this term is new to the user too, we've always talked about "results". Violations exist in Prism but they're "Contract Violations", so lets try and avoid this term being advertised too
- Exceptions - usually means `throw new Exception` so this might cause confusion
- Ignore - Even if its run internally, this thing doesn't exist to the user because it is completely ignored
- Skipped - something should have happened but we didnt do it, maybe mention it though to let them know - like a skipped test in Jest

We want to ignore. Ignore a rule, ignore a file, pretend it doesn't exist. No output, no warning, no nothing anywhere.

Reply:
@philsturgeon I agree. Let's not extend the scope of that PR. Once this is merged, I'll open a dedicated issue on that topic.